### PR TITLE
chore: fixes error alerts from lgtm tool 01

### PR DESCRIPTION
**Pull request by paschalidi**
hey @JoakimSM as we talked about some time ago. 

## What is this PR?
I am opening this PR to fix some lgmt errors. In this series of PRs the goal is to have `eslint` checking our PRs so that if any error occurs that is `eslint` related the github checkers will spot it for us. This will include only errors and not warnings.

I am not intending to change any `eslint` rules, I will simply fix the `errors` that are being reported if any. This step is necessary so that we can then add the `eslint` check in our CI/CD pipeline.  `eslint` then will be checking our codebase for any errors and if there are none then it will give us a success or else will throw an error. The motivation is that we will have a more cohesive style on the code and that devs will not spent time looking into linting errors on PRs and will focus on reviewing the logic of a feature, the readability and so on. 

## What is he bigger picture of the stacked PRs. 

This is gonna be 4 stacked PRs. 

 - The first will fix most of lgtm errors
 - The second will fix most of lgmt warnings
 - The third will fix eslint errors 
 - The fourth will add a script on our pipeline to check for `eslint` errors. Something between the lines `eslint src/*.js`

Please let me know if I could do anything differently.

### CODE REVIEW

File: src/core_modules/capture-core/components/FiltersForTypes/Numeric/NumericFilter.component.js [hunk 1]
```diff
@@ -96,8 +96,7 @@ class NumericFilter extends Component<Props> implements UpdatableFilterContent<V
     }
 
     static isFilterValid(minValue?: ?string, maxValue?: ?string, type: $Values<typeof elementTypes>) {
-        if (!(NumericFilter.validateField(minValue, type).isValid &&
-            NumericFilter.validateField(minValue, type).isValid)) {
+        if (!NumericFilter.validateField(minValue, type).isValid) {
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/FiltersForTypes/Numeric/NumericFilter.component.js:99]:
```suggestion
        if (!NumericFilter.validateField(minValue, type).isValid) || !NumericFilter.validateField(maxValue, type).isValid {
```
Thanks for finding this 😊 The second test should be maxValue

File: src/core_modules/capture-core/components/Pages/NewEvent/SelectionsComplete/SelectionsComplete.component.js [hunk 2]
```diff
@@ -23,17 +23,7 @@ type Props = {
     },
 };
 
-type State = {
-    discardWarningOpen: boolean,
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/Pages/NewEvent/SelectionsComplete/SelectionsComplete.component.js:27]: haha, what's up with the discardWarningOpen.. good riddance 👍

File: src/core_modules/capture-core/components/QuickSelector/Program/CategorySelector.component.js [hunk 3]
```diff
@@ -82,6 +82,7 @@ class CategorySelector extends React.Component<Props, State> {
             prevOrgUnitId: null,
         };
         this.onSelectSelector = makeOnSelectSelector();
+        // todo you cannot setState from this component (lgtm report)
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/QuickSelector/Program/CategorySelector.component.js:85]: Don't remember why I did this from the top of my head. Migth be worth looking into at some point.

**Comment by paschalidi** [comment]
The bug is fixed! Check it out @JoakimSM

**Pull request merged**